lib/pos_snapshot_entries.go

```diff
@@ -688,188 +688,169 @@ type SnapshotStakeMapKey struct {
 	StakerPKID            PKID
 }
 
-// This is a bare bones in-memory only construct used to capture the ValidatorPKID,
-// StakerPKID, and StakeAmountNanos from a StakeEntry that has been snapshotted. We
-// define a new type here rather than re-using the StakeEntry type to reduce the risk
-// of bugs. The StakeEntry type has additional fields (ex: RewardMethod, ExtraData)
-// that are not snapshotted.
-type SnapshotStakeEntry struct {
-	SnapshotAtEpochNumber uint64
-	StakerPKID            *PKID
-	ValidatorPKID         *PKID
-	StakeAmountNanos      *uint256.Int
+func NewSnapshotStakeMapKey(stakeEntry *StakeEntry, snapshotAtEpochNumber uint64) *SnapshotStakeMapKey {
```

Comment:
Typically for map keys, we just return by value instead of by pointer. Any reason we want a pointer here?

Reply:
No reason. We can return by value if that's less error prone!